reaktive/src/commonMain/kotlin/com/badoo/reaktive/single/Filter.kt

```diff
@@ -0,0 +1,25 @@
+package com.badoo.reaktive.single
+
+import com.badoo.reaktive.base.ErrorCallback
+import com.badoo.reaktive.base.Observer
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.base.tryCatch
+import com.badoo.reaktive.maybe.Maybe
+import com.badoo.reaktive.maybe.maybeUnsafe
+
+fun <T> Single<T>.filter(predicate: (T) -> Boolean): Maybe<T> =
+    maybeUnsafe { observer ->
+        subscribeSafe(
```

Comment:
`subscribeSafe` will call `observer.onError` in case of exception during subscription. No callbacks should be called before `obsrever.onSubscribe`. Use `DisposableWrapper` trick. Same mistake I made in `Maybe.filter`, please fix. And some tests would be good.

Reply:
> And some tests would be good (perhaps generic tests to cover such cases everywhere).

Not sure how to implement it, let's move it to another PR.